docking.js

```diff
@@ -860,60 +863,67 @@ var DockedDash = GObject.registerClass({
     }
 
     /**
-     * handler for mouse pressure sensed
+     * Returns whether the global pointer is considered inside of the dock
+     * area or not.
      */
-    _onPressureSensed() {
-        if (Main.overview.visibleTarget)
-            return;
+    _isPointerInZone() {
+        let [x, y, mods] = global.get_pointer();
 
-        // In case the mouse move away from the dock area before hovering it, in such case the leave event
-        // would never be triggered and the dock would stay visible forever.
-        let triggerTimeoutId =  GLib.timeout_add(GLib.PRIORITY_DEFAULT, 250, () => {
-            triggerTimeoutId = 0;
-
-            let [x, y, mods] = global.get_pointer();
-            let shouldHide = true;
             switch (this._position) {
             case St.Side.LEFT:
                 if (x <= this.staticBox.x2 &&
                     x >= this._monitor.x &&
                     y >= this._monitor.y &&
                     y <= this._monitor.y + this._monitor.height) {
-                    shouldHide = false;
+                    return true;
                 }
                 break;
             case St.Side.RIGHT:
                 if (x >= this.staticBox.x1 &&
                     x <= this._monitor.x + this._monitor.width &&
                     y >= this._monitor.y &&
                     y <= this._monitor.y + this._monitor.height) {
-                    shouldHide = false;
+                    return true;
                 }
                 break;
             case St.Side.TOP:
                 if (x >= this._monitor.x &&
                     x <= this._monitor.x + this._monitor.width &&
                     y <= this.staticBox.y2 &&
                     y >= this._monitor.y) {
-                    shouldHide = false;
+                    return true;
                 }
                 break;
             case St.Side.BOTTOM:
                 if (x >= this._monitor.x &&
                     x <= this._monitor.x + this._monitor.width &&
                     y >= this.staticBox.y1 &&
                     y <= this._monitor.y + this._monitor.height) {
-                    shouldHide = false;
+                    return true;
                 }
+                break;
             }
-            if (shouldHide) {
+
+        return false;
+    }
```

Comment:
Moved this big boy out so it can be used both in `_onPressureSensed` and `_hoverChanged`.